pkg/operator/ceph/cluster/controller.go

```diff
@@ -265,22 +266,28 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 
 			monSecret, clusterFSID, err := r.clusterController.getCleanUpDetails(cephCluster.Namespace)
 			if err != nil {
-				return reconcile.Result{}, errors.Wrap(err, "failed to get mon secret, no cleanup")
+				logger.Warningf("failed to get mon secret. Skip cluster cleanup and remove finalizer. %v", err)
+				doCleanup = false
 			}
-			cephHosts, err := r.clusterController.getCephHosts(cephCluster.Namespace)
-			if err != nil {
-				return reconcile.Result{}, errors.Wrapf(err, "failed to find valid ceph hosts in the cluster %q", cephCluster.Namespace)
+
+			if doCleanup {
+				cephHosts, err := r.clusterController.getCephHosts(cephCluster.Namespace)
+				if err != nil {
+					return reconcile.Result{}, errors.Wrapf(err, "failed to find valid ceph hosts in the cluster %q", cephCluster.Namespace)
```

Comment:
Don't we have a similar problem here?

Reply:
I think we can still call the requestClusterDelete function and delete the PVCs even if doCleanup is false. I don't see this method accessing the ceph cluster. Its just deleting the PVCs.  So we can remove the check on line 273.  What do you think ?

Reply:
I don't think we will. We are trying to avoid doing any clean when the cluster is not accessible.  Line 267 tries to load the cluster by fetching the mon secret. So if that works, we can be assured that the cluster is accessible and clean up can be performed. `getCephHosts` is just using the k8s api server to fetch the daemons.